pyccel/codegen/printing/ccode.py

```diff
@@ -893,26 +893,16 @@ def _print_MathFunctionBase(self, expr):
                 self._additional_imports.add('math')
         args = []
         for arg in expr.args:
-            if arg.dtype != expr.dtype:
-                cast_func = python_builtin_datatypes[str_dtype(expr.dtype)]
-                args.append(self._print(cast_func(arg)))
+            if arg.dtype != NativeReal() and not func_name.startswith("pyc"):
+                args.append(self._print(PythonFloat(arg)))
             else:
                 args.append(self._print(arg))
         code_args = ', '.join(args)
+        if expr.dtype == NativeInteger():
+            cast_type = self.find_in_dtype_registry('int', default_precision['int'])
+            return '({0}){1}({2})'.format(cast_type, func_name, code_args)
```

Comment:
Does this need doing for all math functions?

Reply:
all maths functions accept real numbers in their arguments and return real numbers as result.

Reply:
ok perfect :+1: